pipeline/src/pipeline.rs

```diff
@@ -107,6 +107,55 @@ enum Artifact<T: FieldElement> {
     Proof(ProofResult<T>),
 }
 
+// These are implementations of specific artifacts we want to retrieve
+// from Pipeline, in a way that allows us to get immutable references
+// to the artifacts.
+impl<T: FieldElement> Artifact<T> {
+    pub fn try_as_asm_string(&self) -> Result<&String, Vec<String>> {
```

Comment:
Why are they all called `try_` if they only panic on failure?

Reply:
true, will remove `try` from all names

Reply:
yea that makes sense

Reply:
I think `to_foobar` returning an `Option` is idiomatic enough.
For example `OsStr::to_str` returns an optional reference to owned data.